### PR TITLE
Feature/Send Email in New Accounts and Refactor Emails

### PR DESCRIPTION
**Pull request by kevinpr007**
# Send Email in New Accounts and Refactor Emails

## Change log: 
+ Modify and Refactor the following list of emails below:
    + Reset Password
    + Confirm Resert Password
 

## Additional changes:
+ Create a new Email Service
+ Create a new Factory for email's object
+ Create a new email when you register a new email account

## Pending changes:
+ Verify the Auth and Register work flow

### CODE REVIEW

File: src/controllers/auth.controller.js [hunk 1]
```diff
@@ -100,7 +112,11 @@ let authController = (authService, userService, templateModel) => {
         })
       }
     }).then(user => {
-      return resetPasswordEmail(user)
+      // Reset Password Email
+      let params = []
```
Comment (Ivan-Marquez) [inline src/controllers/auth.controller.js]: Hi @kevinpr007,

If it's empty, then there's no use to send it as a parameter. If we need to send parameters later on, we'll do a refactor.

File: src/services/emails/confirm-reset-password.js [hunk 2]
```diff
@@ -1,29 +1,20 @@
...
+module.exports = params => {
+  let getTemplate = () => {
+    let template = {
+      subject: 'Style Point: Reset your password',
+      html: `
         <p>
             You are receiving this because you (or someone else) have requested the reset of the password for your account.
             <br>Please click on the following link, or paste this into your browser to complete the process:<br><br>
-            http://${host}/api/reset-password/${user.resetPasswordToken}<br><br>
+            http://${params[0]}/api/reset-password/${params[1]}<br><br>
```
Comment (Ivan-Marquez) [inline src/services/emails/confirm-reset-password.js]: For better code readability, I'd rather see the properties used inside each string template. Using `params[]` makes you go through all the code just to find out what you're sending as parameters.

File: src/controllers/auth.controller.js (hunk 1 above)
Comment (kevinpr007) [inline src/controllers/auth.controller.js]: I did the fix

File: src/services/emails/confirm-reset-password.js (hunk 2 above)
Comment (kevinpr007) [inline src/services/emails/confirm-reset-password.js]: I all ready changed the code

**Pull request merged**